packages/wrangler/src/pages/build-env.ts

```diff
@@ -33,25 +37,37 @@ export const Handler = async (args: PagesBuildEnvArgs) => {
 		throw new FatalError("No outfile specified");
 	}
 
+	logger.log(
+		"Checking for configuration in a wrangler.toml configuration file (BETA)\n"
+	);
+
 	const configPath = path.resolve(args.projectDir, "wrangler.toml");
-	// Fail early if the config file doesn't exist
 	if (!existsSync(configPath)) {
-		throw new FatalError(
-			"No Pages config file found",
-			EXIT_CODE_NO_CONFIG_FOUND
-		);
+		logger.debug("No wrangler.toml configuration file found. Exiting.");
+		process.exitCode = EXIT_CODE_NO_CONFIG_FOUND;
+		return;
 	}
-	logger.log("Reading build configuration from your wrangler.toml file...");
 
-	const config = readConfig(
-		configPath,
-		{
-			...args,
-			// eslint-disable-next-line turbo/no-undeclared-env-vars
-			env: process.env.PAGES_ENVIRONMENT,
-		},
-		true
-	);
+	logger.log("Found wrangler.toml file. Reading build configuration...");
+
+	let config: Omit<Config, "pages_build_output_dir"> & {
+		pages_build_output_dir: string;
+	};
+	try {
+		config = readConfig(
+			configPath,
+			{
+				...args,
+				// eslint-disable-next-line turbo/no-undeclared-env-vars
+				env: process.env.PAGES_ENVIRONMENT,
+			},
+			true
+		);
+	} catch (err) {
+		logger.debug("wrangler.toml file is invalid. Exiting.");
+		process.exitCode = EXIT_CODE_INVALID_PAGES_CONFIG;
```

Comment:
Why did we do this format instead of throwing a fatal error with a custom error code like we do for other errors communicated to Pages CI? e.g. https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/config/index.ts#L93

Reply:
We don't exit with `FatalError`s any more (that one you linked is caught by this catch handler)